### ISSUE TITLE
Add ability to track `data-src` changes

### ISSUE THREAD

**Issue by yuri-sergiichuk** [verlok/lazyload#361]
**Is your feature request related to a problem? Please describe.**
I have a case when the `data-src` is filled dynamically and I'm not in control when exactly it is going to be filled. Also, it may change from time-to-time.
With the current implementation, if the `data-src` is not yet available, the library just marks the image as `loaded` and forgets about it.

**Describe the solution you'd like**
In such a scenario it'd be useful to have an ability to start the actual download only when the 
`data-src` is available or is changed.

**Describe alternatives you've considered**
I'm tracking the availability of the `data-src` on my side and use `update()` method in order to force the LazyLoad to re-check if there are any images to download.
Also, in order to re-download the image after the `data-src` URL change, it is required to clean up the `was-processed` attribute.

**Comment by verlok** [comment]
Hi there,
thank you for opening this. 

The `elements_selector` option is a CSS selector and it defaults to `"img"`. Have you tried to pass a selector which only considers images with a `data-src`. For example, you could pass `img[data-src]`. 

You would still need to call the `update` method to check if new elements received the attribute, but you wouldn’t need to clean the `data-was-processed` attribute. 

If you manage to get a notification or callback from the JS that set the `data-src`, you could call `update` only once.

**Comment by yuri-sergiichuk** [comment]
Hi @verlok. Thanks for the ide with a custom selector, but it will not help with the `data-was-processed`, cause I still have to clean up the attribute whenever the `data-src` changes.

I guess what I was trying to suggest is to integrate such an `observation` functionality into the library itself. I've glanced at the Mutation Observer API and it looks promising. Probably, I'll go with smth like that for my purposes, but still, I guess it might be a nice-to-have feature for the library.

**Comment by RaZeR-RBI** [comment]
Stumbled upon the same issue, which seems to be also appearing in #358.
I'm using Svelte in a scenario where existing ```img``` tags' ```data-src``` attributes are being changed because of pagination logic.

For those who are looking for a possible workaround for now, you can try to use the ```load(element, force)``` [function](https://github.com/verlok/lazyload/blob/cab8289d853afb71cae0a0bfcd0683b4ab716a9a/src/lazyload.js#L44) with ```force = true``` to force image reloading.

Here is how I'm using it in a Svelte component (LazyLoad library is being loaded through ```<script src...``` and one global instance is used, so YMMV):
###### component.svelte
```svelte
<script>
import { afterUpdate } from 'svelte';
import lazyload from './lazyload.js';

export let url;

let element;
// this callback is being called after the DOM update
afterUpdate(() => {
    lazyload(element);
);
</script>

<img
    class='lazy'
    data-src={url}
    bind:this={element}
/>
```

###### lazyload.js
```js
export default function (elem) {
	if (!window.ll) {
		window.ll = new LazyLoad({
			elements_selector: '.lazy'
		});
	} else {
		window.ll.update();
	}
	if (!!elem) window.ll.load(elem, true);
}
```

**Comment by verlok** [comment]
Sorry @yuri-sergiichuk ,
I just won't implement a `MutationObserver` in this library for some specific cases. 
The logic should be developed outside of the library, then you could use LazyLoad's `load(element, true)` (the second parameter is `force`) to load it again.
